### PR TITLE
[11.0][ADD] contract_variable_qty_timesheet: New module

### PR DESCRIPTION
**Pull request by carlosdauden**
@Tecnativa

### CODE REVIEW

File: contract_variable_qty_timesheet/README.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.png
+   :target: https://www.gnu.org/licenses/agpl
+   :alt: License: AGPL-3
+
+===============================
+Contract Variable Qty Timesheet
+===============================
+
+This module extends the functionality of contract_variable_quantity adding
+formulas to allow invoice lines from Timesheet (analytic lines)
```
Comment (pedrobaeza) [inline contract_variable_qty_timesheet/README.rst]: s/to allow invoice/to allow to

Comment (pedrobaeza) [inline contract_variable_qty_timesheet/README.rst]: s/formulas/several variable quantity formulas

File: contract_variable_qty_timesheet/README.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to Invoicing > Sales > Contracts and select or create a new contract.
+#. Check *Generate recurring invoices automatically*.
+#. Add a new recurring invoicing line.
+#. Select "Variable quantity" in column "Qty. type".
+#. Select one formula.
```
Comment (pedrobaeza) [inline contract_variable_qty_timesheet/README.rst]: Select "Tasks Timesheets" or "Analytic Same Product" depending on your need.

File: contract_variable_qty_timesheet/data/contract_line_qty_formula_data.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2018 Tecnativa - Carlos Dauden
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+<record id="contract_line_qty_formula_timesheet" model="contract.line.qty.formula">
+    <field name='name'>Tasks Timesheet</field>
```
Comment (pedrobaeza) [inline contract_variable_qty_timesheet/data/contract_line_qty_formula_data.xml]: Task Timesheets

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Don't include generic SVG icon.

I miss a general formula for invoicing all timesheets (project_id != False). Maybe even I would replace "Task Timesheets" by this one.

**Comment by carlosdauden** [comment]
Changes done

**Comment by pedrobaeza** [comment]
> I miss a general formula for invoicing all timesheets (project_id != False). Maybe even I would replace "Task Timesheets" by this one.

This comment is unattended.

**Comment by carlosdauden** [comment]
I don't understand.
I think the formula is fine

**Comment by carlosdauden** [comment]
Project Timesheets formula added

**Review by pedrobaeza — APPROVED** [review]
:clap:

**Review by feketemihai — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I'm seeing that variable qty formula name is not translatable and it should be for allowing to have the text in user language. I have added this possibility in a direct commit as it doesn't harm anything: https://github.com/OCA/contract/commit/133e88c7a0cbf9f5169d0e8a0042857568e095c1, so please take that new branch and add Spanish translations for the formulas.

**Comment by carlosdauden** [comment]
Translated

**Comment by pedrobaeza** [comment]
Great, we can merge this one.

File: contract_variable_qty_timesheet/README.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+=====
+
+To use this module, you need to:
+
+#. Go to Invoicing > Sales > Contracts and select or create a new contract.
+#. Check *Generate recurring invoices automatically*.
+#. Add a new recurring invoicing line.
+#. Select "Variable quantity" in column "Qty. type".
+#. Select "Project Timesheets", "Tasks Timesheets" or "Analytic Same Product"
+   depending on your need.
```
Comment (rafaelbn) [inline contract_variable_qty_timesheet/README.rst:23]: - This 3 types of formulas are not obvious for a user, in my opinion I think we mus explain a little bit.
- There is a use case that make this not work: if you follow this steps you will never have a Project created automatically so this doesn't work.

Comment (pedrobaeza) [reply contract_variable_qty_timesheet/README.rst:23]: > There is a use case that make this not work: if you follow this steps you will never have a Project created automatically so this doesn't work.

This is out of scope of this module.

**Pull request merged**